irohad/subscription/subscriber_impl.hpp

```diff
@@ -77,14 +76,28 @@ namespace iroha::subscription {
     /// Stored notification callback
     CallbackFnType on_notify_callback_;
 
-   public:
     template <typename... SubscriberConstructorArgs>
     SubscriberImpl(SubscriptionEnginePtr const &ptr,
                    SubscriberConstructorArgs &&... args)
         : next_id_(0ull),
           engine_(ptr),
           object_(std::forward<SubscriberConstructorArgs>(args)...) {}
 
+   public:
+    template <typename... SubscriberConstructorArgs>
+    static std::shared_ptr<SubscriberImpl> create(
+        SubscriptionEnginePtr const &ptr,
+        SubscriberConstructorArgs &&... args) {
+      struct Resolver : SubscriberImpl {
```

Comment:
Why is this resolver required?

Reply:
You can't make ```make_shared``` as a friend. This resolver is needed to call private ctor.